terminal_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package ansiterm
+
+import (
+	"io/ioutil"
+	"os"
+
+	gc "gopkg.in/check.v1"
+)
+
+type colorWriterSuite struct{}
+
+var _ = gc.Suite(&colorWriterSuite{})
+
+func (s *colorWriterSuite) TestNoColor(c *gc.C) {
+	file, err := ioutil.TempFile("", "")
+	c.Assert(err, gc.IsNil)
+
+	os.Setenv("NO_COLOR", "true")
+
+	writer, ok := colorEnabledWriter(file)
+	c.Assert(ok, gc.Equals, false)
+	c.Assert(writer, gc.Equals, file)
+}
+
+func (s *colorWriterSuite) TestTruthy(c *gc.C) {
+	tests := []struct {
+		Value    string
+		Expected bool
+	}{{
+		Value:    "y",
+		Expected: true,
+	}, {
+		Value:    "Y",
+		Expected: true,
+	}, {
+		Value:    "TRUE",
+		Expected: true,
+	}, {
+		Value:    "true",
+		Expected: true,
+	}, {
+		Value:    "1",
+		Expected: true,
+	}, {
+		Value:    "2",
+		Expected: true,
+	}, {
+		Value:    "10000",
+		Expected: true,
+	}, {
+		Value:    "f",
```

Comment:
it seems odd you check for 'f' but no 't'. I would have expected "yes" and/or "no", but the actual org seems to just say "if the ENV var exists".